File: 3853_merge_close_characters/solution.php
<?php
// LeetCode 3853 - Merge Close Characters
// https://leetcode.com/problems/merge-close-characters/

class Solution {
    function mergeCharacters($s, $k) {
        $last = [];
        $ans = '';
        $n = strlen($s);
        for ($i = 0; $i < $n; $i++) {
            $c = $s[$i];
            $cur = strlen($ans);
            if (isset($last[$c]) && $cur - $last[$c] <= $k) continue;
            $ans .= $c;
            $last[$c] = $cur;
        }
        return $ans;
    }
}

File: 3853_merge_close_characters/Solution.swift
// LeetCode 3853 - Merge Close Characters
// https://leetcode.com/problems/merge-close-characters/

class Solution {
    func mergeCharacters(_ s: String, _ k: Int) -> String {
        var last = [Character: Int]()
        var ans = [Character]()
        for c in s {
            let cur = ans.count
            if let p = last[c], cur - p <= k { continue }
            ans.append(c)
            last[c] = cur
        }
        return String(ans)
    }
}
